cmd/component-codegen/cmd/generator/makecmd.go

```diff
@@ -0,0 +1,23 @@
+package generator
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/sirupsen/logrus"
+)
+
+// runMakeCommand executes necessary Makefile commands for updating generated code and manifests.
+func RunMakeCommand(logger *logrus.Logger) error {
```

Comment:
This feels strange, to have `make` invoke `component-codegen`, and then `component-codegen` invoke `make`. It might be simpler to just have `make` run the necessary targets after the new files have been scaffolded?

Also, the `build` target shouldn't be necessary here. I guess you've added it because of `fmt` and `vet`, right? Ideally the templates should be formatted in a way that `fmt` is happy with the output, without having to be post-processed. WDYT?

Reply:
That makes sense.
So now make takes care of the `generate manifests api-docs`.
As for the `go fmt` I believe it is necessary as the `types.go.tmpl` has something like 
```
type DSC{{ .Component }} struct {
	common.ManagementSpec  `json:",inline"`
	{{ .Component }}CommonSpec `json:",inline"`
```
The Component's(Name) length is subjected to change and as a result the struct is realligned by go.
So I guess using the fmt(PHONY) of make is okay . WDYT?